src/tiled/tileseteditor.cpp

```diff
@@ -465,7 +465,28 @@ void TilesetEditor::addTiles()
         Tile *newTile = new Tile(tileset->takeNextTileId(), tileset);
         newTile->setImage(loadedFile.image);
         newTile->setImageSource(loadedFile.imageSource);
-        tiles.append(newTile);
+
+        // If the tile is already in the tileset, warn user and confirm addition
+        bool addImage = true;
+        const QMap<int, Tile*> addedTiles = mCurrentTilesetDocument->tileset().data()->tiles();
+        for(auto tile : addedTiles) {
+            if(tile->imageSource() == newTile->imageSource()) {
+                QMessageBox warning(QMessageBox::Warning,
+                            tr("Add Tiles"),
+                            tr("Tile already exists in the Tileset!"),
+                            QMessageBox::Yes | QMessageBox::No,
```

Comment:
I think that's fine. Indeed the checkbox can only apply to the current operation.



> seemed more appropriate than "Don't show this again" from Windows Explorer.

I was referring to the Explorer because I assumed would have a better text than that. I just tried it, and actually it solves the problem without a checkbox:

![2017-03-03 10_27_17-replace or skip files](https://cloud.githubusercontent.com/assets/531764/23545598/528a4bb6-fffc-11e6-802b-2efd6b91ad0d.png)

That is getting really fancy though, so I won't ask you to go that far. I think it's no longer worth the time investment at that point.